packages/engine/src/scene/components/ObjectGridSnapComponent.ts

```diff
@@ -23,136 +23,145 @@ All portions of the code written by the Ethereal Engine team are Copyright © 20
-  const lineSegmentList: Vector3[] = []
-  const size = new Vector3()
-  bbox.getSize(size)
-
-  // Create grid lines for each face of the bounding box
-
-  // Front and back faces (parallel to XY plane)
-  const zFront = bbox.min.z
-  const zBack = bbox.max.z
-  for (let j = 0; j <= density; j++) {
-    const segmentFraction = j / density
-    const x = bbox.min.x + segmentFraction * size.x
-    const y = bbox.min.y + segmentFraction * size.y
-
-    lineSegmentList.push(new Vector3(x, bbox.min.y, zFront), new Vector3(x, bbox.max.y, zFront))
-    lineSegmentList.push(new Vector3(x, bbox.min.y, zBack), new Vector3(x, bbox.max.y, zBack))
-    lineSegmentList.push(new Vector3(bbox.min.x, y, zFront), new Vector3(bbox.max.x, y, zFront))
-    lineSegmentList.push(new Vector3(bbox.min.x, y, zBack), new Vector3(bbox.max.x, y, zBack))
-  }
+export const BoundingBoxHelperComponent = defineComponent({
+  name: 'BoundingBoxHelperComponent',
 
-  // Top and bottom faces (parallel to XZ plane)
-  const yTop = bbox.max.y
-  const yBottom = bbox.min.y
-  for (let j = 0; j <= density; j++) {
-    const segmentFraction = j / density
-    const x = bbox.min.x + segmentFraction * size.x
-    const z = bbox.min.z + segmentFraction * size.z
-
-    lineSegmentList.push(new Vector3(x, yTop, bbox.min.z), new Vector3(x, yTop, bbox.max.z))
-    lineSegmentList.push(new Vector3(x, yBottom, bbox.min.z), new Vector3(x, yBottom, bbox.max.z))
-    lineSegmentList.push(new Vector3(bbox.min.x, yTop, z), new Vector3(bbox.max.x, yTop, z))
-    lineSegmentList.push(new Vector3(bbox.min.x, yBottom, z), new Vector3(bbox.max.x, yBottom, z))
-  }
+  onInit: (entity) => {
+    return {
+      name: 'bounding-box-helper',
+      bbox: new Box3(),
+      matrixWorld: new Matrix4().identity(),
+      density: 2,
+      entity: undefined as undefined | Entity
+    }
+  },
 
-  // Left and right faces (parallel to YZ plane)
-  const xLeft = bbox.min.x
-  const xRight = bbox.max.x
-  for (let j = 0; j <= density; j++) {
-    const segmentFraction = j / density
-    const y = bbox.min.y + segmentFraction * size.y
-    const z = bbox.min.z + segmentFraction * size.z
-
-    lineSegmentList.push(new Vector3(xLeft, y, bbox.min.z), new Vector3(xLeft, y, bbox.max.z))
-    lineSegmentList.push(new Vector3(xRight, y, bbox.min.z), new Vector3(xRight, y, bbox.max.z))
-    lineSegmentList.push(new Vector3(xLeft, bbox.min.y, z), new Vector3(xLeft, bbox.max.y, z))
-    lineSegmentList.push(new Vector3(xRight, bbox.min.y, z), new Vector3(xRight, bbox.max.y, z))
-  }
-  for (let i = 0; i < lineSegmentList.length; i++) {
-    lineSegmentList[i].applyMatrix4(matrixWorld)
+  onSet: (entity, component, json) => {
+    if (!json || !json.bbox || !json.bbox.isBox3) throw new Error('BoundingBoxHelperComponent: Requires Box3')
+    component.bbox.set(json.bbox)
+
+    if (matchesMatrix4.test(json.matrixWorld)) component.matrixWorld.set(json.matrixWorld)
```

Comment:
if this matrixWorld is only being used internally, we don't need/want a deserializer for it. furthermore, is this distinct from the transform matrixWorld? if not, we should remove it